CrossSectionAnalysis/CrossSectionAnalysis.py

```diff
@@ -1263,32 +1215,54 @@ def updateOutputTable(self, inputCenterline, outputTable):
-    elif self.outputPlotSeriesType == DIAMETER_STENOSIS:
-        outputPlotSeries.SetYColumnName(DIAMETER_STENOSIS_ARRAY_NAME)
-    elif self.outputPlotSeriesType == SURFACE_AREA_STENOSIS:
-        outputPlotSeries.SetYColumnName(SURFACE_AREA_STENOSIS_ARRAY_NAME)
-    else:
-      pass
-    outputPlotSeries.SetPlotType(slicer.vtkMRMLPlotSeriesNode.PlotTypeScatter)
-    outputPlotSeries.SetMarkerStyle(slicer.vtkMRMLPlotSeriesNode.MarkerStyleNone)
-    outputPlotSeries.SetColor(0, 0.6, 1.0)
+  def _createOutputPlotSeries(self, name, yColumnName, unit, color = (0.0, 0.6, 1.0)):
+    plotSeriesNode = slicer.mrmlScene.AddNewNodeByClass("vtkMRMLPlotSeriesNode")
+    # It will be weird to get nodes in the python console;
+    # It is nevertheless useful since we can't use a Y axis title.
+    fullName = "{nameOfArray} ({unitOfMeasurement})".format(nameOfArray=yColumnName, unitOfMeasurement=unit)
+    plotSeriesNode.SetName(fullName)
+    plotSeriesNode.SetXColumnName(DISTANCE_ARRAY_NAME)
+    plotSeriesNode.SetYColumnName(yColumnName)
+    plotSeriesNode.SetPlotType(slicer.vtkMRMLPlotSeriesNode.PlotTypeScatter)
+    plotSeriesNode.SetMarkerStyle(slicer.vtkMRMLPlotSeriesNode.MarkerStyleNone)
+    plotSeriesNode.SetColor(color)
+    plotSeriesNode.SetAndObserveTableNodeID(self.outputTableNode.GetID())
+    # Make sure the plot is in the chart
+    if not self.outputPlotChartNode.HasPlotSeriesNodeID(plotSeriesNode.GetID()):
+      self.outputPlotChartNode.AddAndObservePlotSeriesNodeID(plotSeriesNode.GetID())
+    return plotSeriesNode
+
+  def updatePlot(self, outputPlotChart, outputTable, name=None):
+    # Recreate all output plot series nodes.
+    self.outputPlotSeriesNodes.clear()
+    self.outputPlotChartNode.RemoveAllPlotSeriesNodeIDs()
+
+    lengthUnit = self.getUnitNodeUnitDisplayString(0.0, "length")
+    areaUnit = self.getUnitNodeUnitDisplayString(0.0, "area")
+
+    self.outputPlotChartNode.SetXAxisTitle("{nameOfDistanceArray} ( {unitOfLength})".format(nameOfDistanceArray=DISTANCE_ARRAY_NAME, unitOfLength=lengthUnit))
+
+    if self.isCenterlineRadiusAvailable():
+      misDiameterSeriesNode = self._createOutputPlotSeries(MIS_DIAMETER_ARRAY_NAME, MIS_DIAMETER_ARRAY_NAME, lengthUnit, MIS_DIAMETER_PLOT_SERIES_COLOR)
+      # self.outputPlotChartNode.SetYAxisTitle(_("Diameter ({unitOfLength})").format(unitOfLength=lengthUnit))
+      self.outputPlotSeriesNodes.append(misDiameterSeriesNode)
+
+    if self.lumenSurfaceNode:
+      ceDiameterSeriesNode = self._createOutputPlotSeries(CE_DIAMETER_ARRAY_NAME, CE_DIAMETER_ARRAY_NAME, lengthUnit, CE_DIAMETER_PLOT_SERIES_COLOR)
+      self.outputPlotSeriesNodes.append(ceDiameterSeriesNode)
+      crossSectionAreaSeriesNode = self._createOutputPlotSeries(CROSS_SECTION_AREA_ARRAY_NAME, CROSS_SECTION_AREA_ARRAY_NAME, areaUnit, CROSS_SECTION_AREA_PLOT_SERIES_COLOR)
+      self.outputPlotSeriesNodes.append(crossSectionAreaSeriesNode)
```

Comment:
We should not display length and area values in the same chart. What would be the meaning of the vertical axis then?

Reply:
Well the units are appended to the plot series name, and this appears in the legend. The Y axis shows a range that varies with the selected series.

![plot_all](https://github.com/user-attachments/assets/107b9127-c775-4a7d-9e99-0f3ce96d5e52)

![plot_single](https://github.com/user-attachments/assets/0a107f21-9825-4d4d-85a8-b9cde94c3766)

Reply:
Diameter, cross-section, and percentage should never appear in the same plot, as they are different quantities. VTK might support dual axis charts, which could be used to display absolute and percentage value; or length and area quantities in one chart. Can you check if VTK indeed supports it? (there is a method to get left, right, top, bottom axis, but I'm not sure if you can set different ranges for them)

Reply:
> Can you check if VTK indeed supports it?
I could not find this support. In any case, only dual axis won't suffice.

> should never appear in the same plot
In that case, the plot rework becomes pointless. I'll limit this PR to the parameter set addition. It will still be easier to work with multiple centerlines.